### ISSUE TITLE
Github and logins

### ISSUE THREAD

**Issue by con-f-use** [andir/npins#100]
Having a difficult time with `npins add github ...`. Why does it prompt me for a github username and password, if the repo I'm trying to add is public? It's especially puzzling because I have git configured to use ssh and because github has discontinued https username+password auth for a long time.

**Comment by piegamesde** [comment]
If you plainly `git clone` the same repo, does it do the same?

**Comment by con-f-use** [comment]
It does not, no.

**Comment by piegamesde** [comment]
Oh well. We really need some debug logging, in which it should show which commands npins is executing. Assuming it is failing on the first command, that would likely be some `git ls-remote` to find out the latest release. Moreover, we really should instruct git to operate in a non-interactive way and to fail instead of prompting.

Some things which would be cool if you tried it out:

- Which version of npins are you using?
- Can you interactively enter some credentials into the prompt? (Doesn't have to be correct ones, just any)
- Does it work for other repos?

**Comment by piegamesde** [comment]
Oh I'm just seeing, you got a typo in the repo name of your `npins add` command. That at least explains why it is failing. However, I still don't understand why you get a prompt, on my machine it fails correctly instead:

```
npins add github nix-community mine-flake
[INFO ] Adding 'mine-flake' …
[ERROR] git ls-remote failed. stderr output:
[ERROR] > remote: Repository not found.
[ERROR] > fatal: repository 'https://github.com/nix-community/mine-flake.git/' not found
[ERROR] > 
Error: Failed to fully initialize the pin

Caused by:
    0: Couldn't fetch the release tags
    1: Failed to list tags for https://github.com/nix-community/mine-flake.git
    2: git ls-remote failed with exit code 128
```

**Comment by con-f-use** [comment]
Oh wow, I'm sorry to have bothered you then. I typoed once and then kept on using the wrong command via bash history. :facepalm: 
I think the prompt is git's default behavior because the repo could be private and git can´t distingish. Could that be?
In any case, according to the nix shell, I was using npins 0.2.4 from nixpkgs. I can enter credentials, yes.

**Comment by piegamesde** [comment]
Yes, that narrows it down. I have a `http` token for GitHub in my `.git-credentials`, so I'm already "logged in" to GitHub and it can show me a 404. If I comment it out (or configure only SSH instead), GitHub wants login to check if the repository may exist. I even get a fancy GUI prompt, with fallback to CLI, though both seem to work.

I'll guess that this is not specific to GitHub and may happen with other forges as well.

Regardless, I'd consider this behavior an error, and the proper thing would be to instruct Git to not use interactive authentication.

**Comment by piegamesde** [comment]
Can you please try again with `GIT_TERMINAL_PROMPT=0`?
